pkg/utils/logger/middleware.go

```diff
@@ -0,0 +1,54 @@
+package logger
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+)
+
+// GinLogger returns a new Gin middleware that performs logging for our JSON APIs using
+// zerolog rather than the default Gin logger which is a standard HTTP logger. Provide
+// the server name (e.g. adminAPI or BFF) to help us parse the logs.
+// NOTE: we previously used github.com/dn365/gin-zerolog but wanted more customization.
+func GinLogger(server string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Before request
+		started := time.Now()
+		path := c.Request.URL.Path
+		if c.Request.URL.RawQuery != "" {
+			path = path + "?" + c.Request.URL.RawQuery
+		}
+
+		// Handle the request
+		c.Next()
+
+		// After request
+		status := c.Writer.Status()
+		logctx := log.With().
+			Str("path", path).
+			Str("ser_name", server).
+			Str("method", c.Request.Method).
+			Dur("resp_time", time.Since(started)).
+			Int("resp_bytes", c.Writer.Size()).
+			Int("status", status).
+			Str("client_ip", c.ClientIP()).
+			Logger()
+
+		// This field requires us to append errors to the Gin context before a 500
+		msg := c.Errors.String()
+		if msg == "" {
+			msg = fmt.Sprintf("%s %s %s %d", server, c.Request.Method, c.Request.URL.Path, status)
+		}
+
+		switch {
+		case status >= 400 && status < 500:
+			logctx.Warn().Msg(msg)
+		case status > 500:
+			logctx.Error().Msg(msg)
+		default:
+			logctx.Info().Msg(msg)
+		}
+	}
+}
```

Comment:
This was basically all that was in `gin-zerolog` Go package -- but the log message in Google Console was "Request" … you had to dive into the JSON to get any information which was annoying. So now the message contains more info including status code, method, and path. Plus we got rid of a dependency!

Reply:
I added the request parameters to the path on line 21 - honestly, this was just copying what gin-zerolog was doing before; do you think we should have the params logged separately from the path?